src/hydrusvideodeduplicator/dedup.py

```diff
@@ -204,29 +232,39 @@ def _find_potential_duplicates(
         video_counter = 0
         with SqliteDict(
             str(DedupeDB.get_db_file_path()), tablename="videos", flag="c", autocommit=True, outer_stack=False
-        ) as hashdb:
+        ) as videos_table:
             current_hash = None
             try:
-                total = len(hashdb)
+                # Make a copy of the video hashes here so we can preserve their order because SqliteDict row order
+                # changes during writes for the farthest search index. This is a bandaid solution.
+                # This assumes SqliteDict row order is preserved when opened and closed, even if it's not preserved
+                # while modifying elements.
+                video_hashes = [video_hash for video_hash in videos_table]
+                total = len(video_hashes)
 
                 with tqdm(
                     dynamic_ncols=True, total=total, desc="Finding duplicates", unit="video", colour="BLUE"
                 ) as pbar:
                     # -1 is all cores, -2 is all cores but one
                     with Parallel(n_jobs=self.job_count) as parallel:
```

Comment:
@ianwal Huh, I thought I saw `return_as='unordered'` make it into a previous release - how come it didn't make it into this release?